src/figuresClasses.ts

```diff
@@ -1,19 +1,102 @@
 export interface Figure {
-
+  shape:string,
+  color:string,
+  getArea():number,
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  a:number;
+
+  b:number;
+
+  c:number;
+
+  shape:string;
+
+  color:string;
 
+  constructor(
+    color:string,
+    a:number,
+    b:number,
+    c:number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Zero value argument');
+    }
+
+    if (Math.max(a, b, c) >= (a + b + c - Math.max(a, b, c))) {
+      throw new Error('Invalid triangle sides');
+    }
+
+    this.shape = 'triangle';
+    this.color = color;
+    this.a = a;
+    this.b = b;
+    this.c = c;
+  }
+
+  getArea():number {
+    const p:number = (this.a + this.b + this.c) / 2;
```

Comment:
```suggestion
    const p = (this.a + this.b + this.c) / 2;
```
the result of calculation will be number so typescript will know about that